packages/std/src/math/decimal.rs

```diff
@@ -696,26 +704,77 @@ mod tests {
-        assert_eq!(Decimal(100000).to_string(), "0.0000000000001");
-        assert_eq!(Decimal(1000000).to_string(), "0.000000000001");
-        assert_eq!(Decimal(10000000).to_string(), "0.00000000001");
-        assert_eq!(Decimal(100000000).to_string(), "0.0000000001");
-        assert_eq!(Decimal(1000000000).to_string(), "0.000000001");
-        assert_eq!(Decimal(10000000000).to_string(), "0.00000001");
-        assert_eq!(Decimal(100000000000).to_string(), "0.0000001");
-        assert_eq!(Decimal(10000000000000).to_string(), "0.00001");
-        assert_eq!(Decimal(100000000000000).to_string(), "0.0001");
-        assert_eq!(Decimal(1000000000000000).to_string(), "0.001");
-        assert_eq!(Decimal(10000000000000000).to_string(), "0.01");
-        assert_eq!(Decimal(100000000000000000).to_string(), "0.1");
+        assert_eq!(
+            Decimal(Uint128::from(1u128)).to_string(),
+            "0.000000000000000001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(10u128)).to_string(),
+            "0.00000000000000001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(100u128)).to_string(),
+            "0.0000000000000001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(1000u128)).to_string(),
+            "0.000000000000001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(10000u128)).to_string(),
+            "0.00000000000001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(100000u128)).to_string(),
+            "0.0000000000001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(1000000u128)).to_string(),
+            "0.000000000001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(10000000u128)).to_string(),
+            "0.00000000001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(100000000u128)).to_string(),
+            "0.0000000001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(1000000000u128)).to_string(),
+            "0.000000001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(10000000000u128)).to_string(),
+            "0.00000001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(100000000000u128)).to_string(),
+            "0.0000001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(10000000000000u128)).to_string(),
+            "0.00001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(100000000000000u128)).to_string(),
+            "0.0001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(1000000000000000u128)).to_string(),
+            "0.001"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(10000000000000000u128)).to_string(),
+            "0.01"
+        );
+        assert_eq!(
+            Decimal(Uint128::from(100000000000000000u128)).to_string(),
+            "0.1"
+        );
```

Comment:
Much harder to review now, but I don't have a better idea

Reply:
I guess we could add something like `impl From<u128> for Decimal<Uint128>` and it would become `Decimal::from(123u128)`?

Reply:
I though about this for a second too but then thought there is no semantic meaning in converting an unsigned integer to a decimal the way we want. If we did that, we had to convert `100u128` to `100.0`.

Reply:
Oh, right. I forgot we're basically poking at internals in tests here.

We can always define some helper function for tests.

Reply:
No worries. Not worth the effort for this one case.